### ISSUE TITLE
Mismatched number of words in CMU-MOSEI paper and CMU_MOSEI_TimestampedWordVectors.csd?

### ISSUE THREAD

**Issue by yiwang454** [A2Zadeh/CMU-MultimodalSDK#263]
Hi Amir @A2Zadeh, 

I downloaded the glove wordvector data from http://immortal.multicomp.cs.cmu.edu/CMU-MOSEI/language/CMU_MOSEI_TimestampedWordVectors.csd (which is the latest one right?)
and counted the number of vectors included by the following code:
```
basic_dict={'COVAREP': '..\\CMU-MultimodalSDK\\cmumosei\\CMU_MOSEI_COVAREP.csd', 
            'OpenFace': '..\\CMU-MultimodalSDK\\cmumosei\\CMU_MOSEI_VisualOpenFace2.csd', 
            'mylabels':'..\\CMU-MultimodalSDK\\cmumosei\\CMU_MOSEI_Labels.csd',
       'WordVec': '..\\CMU-MultimodalSDK\\cmumosei\\CMU_MOSEI_TimestampedWordVectors.csd'}
WordVec = basic_dataset.computational_sequences['WordVec']
all_id_wordvec = list(WordVec.keys())
count_words = 0
for ID in all_id_wordvec:
    count_words += WordVec[ID]['features'].shape[0]
``` 
And got 1084005 for the count_words, but in the CMU_MOSEI paper the number of words are 447143 (which is quite different), so I'm wondering why this is the case? (1084005 is also the count of word number from file CMU_MOSEI_TimestampedWords.csd)

Btw even if I excluded the b'sp' in  CMU_MOSEI_TimestampedWords.csd, there's still 893054 words and I don't quite understand why it's twice the number than in paper.

**Comment by yiwang454** [comment]
Btw another question. For seq20 data from http://immortal.multicomp.cs.cmu.edu/raw_datasets/processed_data/cmu-mosei/seq_length_20/, (15290, 20, 74) is the shape of data in audio_train.h5, but I'm confused about what does 20 mean here. 15290 is the number of utterance(sentence) and 74 is the vector dimension of a datapoint, so does 20 means you cut one sentences into 20 clips and extract features? Then how to clip all sentences into 20 parts if the sentences are of different lengths (in time and in number of words)?

**Comment by yiwang454** [comment]
And another question
I printed out the number of videos for each modal and found that for the up to date dataset some videos missed labels. In this case are those videos ignored in alignment?
```
3836 COVAREP
================================================================================
3837 OpenFace
================================================================================
3293 mylabels
================================================================================
3837 Facet
================================================================================
3837 WordVec
================================================================================
3837 Word
================================================================================
3837 Phone
```

**Comment by A2Zadeh** [comment]
Hello @yiwang454 

Wow great detailed look at the dataset :)

First, let's talk about the words. The number reported in the paper talks about the supervised portion of the dataset. I also believe we report words that occur at least 5 or 10 times. Not sure what the exact number is on that. I am not ruling out a mistake on our side either. In that case, the good news is the dataset may be bigger than originally thought :)

Second, you are absolutely right about the sequence 20. Note we add the padding to the left. 

Third, yeah a few videos are missing their featuers or labels. This should be automatically removed by the SDK itself with a warning. 

Let me know if you had any other questions.

**Comment by yiwang454** [comment]
Hi Amir @A2Zadeh, 

About the word, I counted all the words regardless of whether they had corresponding labels so now I got the reason why it's much larger than the number stated in the paper. (I probably won't go further on checking the number cuz I asked about the number just because I was not sure if I had been using the correct version of dataset and now I'm clear.)

Thank you so much for answering those three questions! I don't have other questions about statistics of the dataset for now.
